pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,228 @@
+type EgressRouterInterface struct {
+	// Arguments specific to the interfaceType macvlan
+	// +kubebuilder:validation:Optional
+	Macvlan MacvlanConfig `json:"macvlan"`
+}
+
+// MacvlanMode defines the different types of modes that are supported for the macvlan interface.
+// source: https://man7.org/linux/man-pages/man8/ip-link.8.html
+type MacvlanMode string
+
+const (
+	// MacvlanModeBridge connects all endpoints directly to each other, communication is not redirected through the physical interface's peer.
+	MacvlanModeBridge MacvlanMode = "Bridge"
+
+	// MacvlanModePrivate does not allow communication between macvlan instances on the same physical interface,
+	// even if the external switch supports hairpin mode.
+	MacvlanModePrivate MacvlanMode = "Private"
+
+	// MacvlanModeVEPA is the Virtual Ethernet Port Aggregator mode. Data from one macvlan instance to the other on the
+	// same physical interface is transmitted over the physical interface. Either the attached switch needs
+	// to support hairpin mode, or there must be a TCP/IP router forwarding the packets in order to allow
+	// communication. This is the default mode.
+	MacvlanModeVEPA MacvlanMode = "VEPA"
+
+	// MacvlanModePassthru mode gives more power to a single endpoint, usually in macvtap mode.
+	// It is not allowed for more than one endpoint on the same physical interface. All traffic will be forwarded
+	// to this endpoint, allowing virtio guests to change MAC address or set promiscuous mode in order to bridge the
+	// interface or create vlan interfaces on top of it.
+	MacvlanModePassthru MacvlanMode = "Passthru"
+)
+
+// MacvlanConfig consists of arguments specific to the macvlan EgressRouterInterfaceType
+type MacvlanConfig struct {
+	// Mode depicts the mode that is used for the macvlan interface; one of Bridge|Private|VEPA|Passthru. The default mode is "Bridge".
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Bridge";"Private";"VEPA";"Passthru"
+	// +kubebuilder:default:="Bridge"
+	Mode MacvlanMode `json:"mode"`
+
+	// Name of the master interface. Need not be specified if it can be inferred from the IP address.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Optional
+	// +optional
+	Master string `json:"master,omitempty"`
+}
+
+// IPConfig contains the configuration for the IP arguments
+// +kubebuilder:validation:Required
+type IPConfig struct {
+	// List of IP addresses to configure on the pod's secondary interface.
+	// +kubebuilder:validation:Required
+	Addresses []EgressRouterAddress `json:"addresses"`
+
+	// List of destinations that the pod is allowed to connect to via this interface. If not provided, the pod can connect to any destination.
+	// Examples:
+	// {1} ["<dst-ip-addr/mask>"] (["203.0.113.25/30"])
+	// {2} ["<src-port> <protocol> <dst-ip-addr/mask>", "<src-port> <protocol> <dst-ip-addr/mask> <dst-port>"] (["80 tcp 203.0.113.25/30", "8080 tcp 203.0.113.26 80", "8443 tcp 203.0.113.26/30 443"])
```

Comment:
So that was fine for the plain text config mode, but if we have a proper JSON config it seems like we should split that out into multiple fields rather than having it all as plain text and then parsing it

Reply:
hmm ok, I'll do a new struct for the `allowedDestinations`, where by we can have the port/hostname/ip fields separately.